### PR TITLE
GCalendar interoperability base

### PR DESCRIPTION
**Pull request by Ef55**
Setup the overall logic for the calendar export.
Implement the UI.

### CODE REVIEW

**Comment by Ef55** [comment]
**Note:** due to the heavy reliance of this code on external UIs (Google SignIn for Android), I sadly cannot get the test coverage higher.

File: app/build.gradle [hunk 1]
```diff
@@ -79,6 +79,14 @@ dependencies {
         exclude group: 'com.android.support'
         exclude group: 'com.mapbox.mapboxsdk' //change when navigation is done
     }
+    // Google login/calendar related
+    implementation 'com.google.android.gms:play-services-auth:19.0.0'
+    implementation 'com.google.android.gms:play-services-auth:19.0.0'
```
Comment (Flechman) [inline app/build.gradle:84]: This line is in duplicate

File: app/src/main/java/com/github/onedirection/interoperability/gcalendar/ExportFragment.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+        Intent signInIntent = googleSignInClient.getSignInIntent();
+        loginGoogle.launch(signInIntent);
+    }
+
+    private void exportEvents(GoogleSignInAccount account) {
+        Account a = Objects.requireNonNull(Objects.requireNonNull(account).getAccount());
+
+        Log.d(LOGCAT_TAG, a.name);
+        // TODO: ask Remi for a method to get all events (in the future ?)
```
Comment (Flechman) [inline app/src/main/java/com/github/onedirection/interoperability/gcalendar/ExportFragment.java:88]: The method `retrieveAll(Storer<T> storer)` in `(Concrete)Database` retrieves all items stored in the database that belong to the given storer. So you can call `retrieveAll(EventStorer.getInstance()` to get all events in the database.

File: app/src/main/java/com/github/onedirection/interoperability/gcalendar/GoogleCalendar.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    // Don't ask my the hows-and-whys of this string, all I know is that it doesn't work otherwise
+    private static final String OAUTH_SCOPE = "oauth2:profile email";
+    private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+    static final String LOGCAT_TAG = "GCalendar";
+
+    private GoogleCalendar() {
+    }
+
+    public static Event toGCalendarEvents(com.github.onedirection.event.Event event) {
```
Comment (Flechman) [inline app/src/main/java/com/github/onedirection/interoperability/gcalendar/GoogleCalendar.java:26]: Provide a small `TODO:` ? Minor thing but just to be consistent

File: app/src/main/java/com/github/onedirection/interoperability/gcalendar/GoogleCalendar.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+        // Example: https://developers.google.com/calendar/v3/reference/events/insert#java
+        // Doc: https://developers.google.com/resources/api-libraries/documentation/calendar/v3/java/latest/
+        // Warning: use EventDateTime::setDatetime, not EventDateTime::setDate
+
+        return null;
+    }
+
+    public static com.github.onedirection.event.Event fromGCalendarEvents(Event event) {
+        return null;
```
Comment (Flechman) [inline app/src/main/java/com/github/onedirection/interoperability/gcalendar/GoogleCalendar.java:37]: Same small `TODO:`

**Review by Flechman — CHANGES_REQUESTED** [review]
Nice approach to the GCalendar. Since you asked for a method to retrieve all events in the db, I'll wait for you to use the method before approving

File: app/src/main/java/com/github/onedirection/interoperability/gcalendar/ExportFragment.java (hunk 2 above)
Comment (Ef55) [reply app/src/main/java/com/github/onedirection/interoperability/gcalendar/ExportFragment.java]: I added a method to _EventQueries_ using the suggested calls to the DB.

File: app/src/main/java/com/github/onedirection/interoperability/gcalendar/GoogleCalendar.java (hunk 3 above)
Comment (Ef55) [reply app/src/main/java/com/github/onedirection/interoperability/gcalendar/GoogleCalendar.java:26]: Added.

File: app/src/main/java/com/github/onedirection/interoperability/gcalendar/GoogleCalendar.java (hunk 4 above)
Comment (Ef55) [reply app/src/main/java/com/github/onedirection/interoperability/gcalendar/GoogleCalendar.java:39]: Added.

File: app/build.gradle (hunk 1 above)
Comment (Ef55) [reply app/build.gradle]: Duplicate removed.

File: app/src/main/java/com/github/onedirection/interoperability/gcalendar/ExportFragment.java (hunk 2 above)
Comment (Flechman) [reply app/src/main/java/com/github/onedirection/interoperability/gcalendar/ExportFragment.java]: Nice

**Review by Flechman — APPROVED** [review]

**Review by justagoodguy-omega — APPROVED** [review]
Seems good

**Pull request merged**